spec/core_functions/selector/unify/compound.hrx

```diff
@@ -94,3 +132,77 @@ a {b: selector.unify(":c", "::d")}
 a {
   b: :c::d;
 }
+
+<===>
+================================================================================
+<===> order/do_not_cross_pseudo_element/pseudo_class_and_element/into_simple/input.scss
+@use "sass:selector";
+a {b: selector.unify(".x::scrollbar:horizontal", ".y")}
+
+<===> order/do_not_cross_pseudo_element/pseudo_class_and_element/into_simple/output.css
+a {
+  b: .x.y::scrollbar:horizontal;
+}
+
+<===>
+================================================================================
+<===> order/do_not_cross_pseudo_element/pseudo_class_and_element/into_pseudo_element/input.scss
+@use "sass:selector";
+a {b: selector.unify("::bar:baz", ":foo")}
+
+<===> order/do_not_cross_pseudo_element/pseudo_class_and_element/into_pseudo_element/output.css
+a {
+  b: :foo::bar:baz;
+}
+
+<===>
+================================================================================
+<===> order/do_not_cross_pseudo_element/pseudo_class_and_element/into_same_pseudo_element_and_different_pseudo_class/input.scss
+@use "sass:selector";
+a {b: selector.unify("::foo:bar", "::foo:baz")}
+
+<===> order/do_not_cross_pseudo_element/pseudo_class_and_element/into_same_pseudo_element_and_different_pseudo_class/output.css
+a {
+  b: ::foo:bar:baz;
+}
+
+<===>
+================================================================================
+<===> order/do_not_cross_pseudo_element/pseudo_class_and_element/into_different_pseudo_element_and_different_pseudo_class/input.scss
+@use "sass:selector";
+a {b: selector.unify("::foo:bar", "::other:baz")}
+
+<===> order/do_not_cross_pseudo_element/pseudo_class_and_element/into_different_pseudo_element_and_different_pseudo_class/output.css
+
+<===>
+================================================================================
+<===> order/do_not_cross_pseudo_element/simple/into_pseudo_class_and_element/input.scss
+@use "sass:selector";
+a {b: selector.unify(".x", ".y::scrollbar:horizontal")}
+
+<===> order/do_not_cross_pseudo_element/simple/into_pseudo_class_and_element/output.css
+a {
+  b: .x.y::scrollbar:horizontal;
+}
+
+<===>
+================================================================================
+<===> order/do_not_cross_pseudo_element/pseudo_element/into_pseudo_class_and_element/input.scss
+@use "sass:selector";
+a {b: selector.unify(":foo", "::bar:baz")}
+
+<===> order/do_not_cross_pseudo_element/pseudo_element/into_pseudo_class_and_element/output.css
+a {
+  b: :foo::bar:baz;
+}
+
+<===>
+================================================================================
+<===> order/cross_pseudo_element/input.scss
+@use "sass:selector";
+a {b: selector.unify(".x", "::y.z")}
```

Comment:
I think we shouldn't test this case, because it's functionally meaningless to begin with.

Reply:
ok, then I think it would make sense to remove the tests where on argument contains two pseudo selectors like `::x::y` since those don't work either IIUC. Let me know if I should add them back tho